src/index.tsx

```diff
@@ -151,8 +151,29 @@ function App() {
     layers.push(subModel.render());
   }
 
+  // This hook checks if the map container parent has a height set, which is
+  // needed to make the map fill the parent container.
+  useEffect(() => {
+    if (mapHeight) return;
+
+    const mapContainer = document.getElementById("map");
+    const mapContainerParent = mapContainer?.parentElement;
+
+    if (mapContainerParent) {
+      // Compute the actual style considering stylesheets, inline styles, and browser default styles
+      const parentStyle = window.getComputedStyle(mapContainerParent);
+
+      // Check if the height is not already set
+      if (!parentStyle.height || parentStyle.height === "0px") {
+        // Set the height to 100% and min-height
+        mapContainerParent.style.height = "100%";
+        mapContainerParent.style.minHeight = "500px";
+      }
+    }
+  }, []);
+
   return (
-    <div style={{ height: mapHeight || 500 }}>
+    <div id="map" style={{ height: mapHeight || "100%" }}>
```

Comment:
I don't think `id="map"` is sufficient because multiple views (of the same or different widgets) could be shown in the same notebook. (This could also be my rendering issue if it works fine for you)